### ISSUE TITLE
[Issue] User Management: Register Passage identity on user creation/activation

### ISSUE THREAD

**Issue by TylerHendrickson** [usdigitalresponse/cpf-reporter#106]
**DRAFT: THIS TICKET IS NOT YET READY FOR WORK**

**Subtask of user story:** User Management https://github.com/usdigitalresponse/cpf-reporter/issues/81

**DOD:** When a user is created or an update to an existing user sets `isActive=true`, an identity is registered in Passage.

**Blocked by issues:**
- https://github.com/usdigitalresponse/cpf-reporter/issues/103

**Blocks these issues:** (none)

**Implementation Details:**
- Update the `User` Prisma model with the following column:
  - `passageId`: text, nullable
- Update the `api/src/lib/auth.ts` module interface to provide an exported function that facilitates [creating a new user identity in Passage](https://docs.passage.id/backend/overview/node#user-management). 
  - These updates must provide a way to retrieve the identifier for the registered user identity.
- Use the `lib/auth` module to create a new Passage user identity so that the following scenarios are satisfied:
  - In the `user` service's `createUser` mutation resolver: when the request's `NewUserInput` input data provides `isActive=true`, a Passage identity is registered with the user's email address and the resulting identifier is set as the `passageId` value for the new user.
   - In the `user` service's `updateUser` mutation resolver: when the request's `UpdateUserInput` input data will set `isActive=true` for a user where currently `isActive=false`, a Passage identity is registered with the user's email address and the resulting identifier is set as the `passageId` value for the new user.

**Comment by as1729** [comment]
TODO: Split create and update functionality into separate tickets. Update/de-activation/re-activation flow is lower priority compared to initial registration of a passage identity.
